Support headers with different header length

JananiKugaa opened the pull request:
## Purpose
 support the message with the different header length

## Goals
Implement the connector to support iso8583 message with header

## Approach

## User stories


## Release note

## Documentation

## Training


## Certification


## Marketing


## Automation tests

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples

## Related PRs


## Migrations (if applicable)


## Test environment
JDK versions-1.8
Operating systems-Ubuntu 16.04
Product- EI-6.1.1

Commit:
Support headers with different header length

menakaj commented on src/main/resources/iso8583_produceMessage/sendMessage.xml:
```diff
@@ -19,5 +19,6 @@
         <class name="org.wso2.carbon.connector.iso8583.ISO8583MessageProducer"/>
+        <script language="js"><![CDATA[mc.getEnvelope().getBody().getFirstElement().detach();]]></script>
     </sequence>
 </template>
```
Shall we add a new line here?

JananiKugaa replied:
Added

Commit:
add a new line

dimslee commented on src/main/java/org/wso2/carbon/connector/iso8583/ISO8583MessageHandler.java:
```diff
@@ -56,28 +57,31 @@ public ISO8583MessageHandler(MessageContext messageContext, String details, Stri
-                if ((message = inFromServer.readLine()) != null) {
-                    unpackResponse(messageContext, message);
-                }
+                Thread.sleep(200);
```
Why did you put 200 sleep here?

JananiKugaa replied:
It will take time to get the response from the back end in the input stream buffer. That's why I put.

dimslee replied:
Please check without it ...

dimslee commented on src/main/java/org/wso2/carbon/connector/iso8583/ISO8583MessageHandler.java:
```diff
@@ -56,28 +57,31 @@ public ISO8583MessageHandler(MessageContext messageContext, String details, Stri
         try {
             outStream = new DataOutputStream(connection.getOutputStream());
-            inFromServer = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            inFromServer = new DataInputStream(connection.getInputStream());
```
Furthermore it is wrong to close connection in this finally block. It should be closed by the method who creates it in it's finally block.

JananiKugaa replied:
resolved

Commit:
Fix from required change

Close the socket connection where it was created, in it's finally block.
Read the byte from input stream with out sleep the thread.
